From 9f59e0a8b70bcd2254a188a2796f1055a659528e Mon Sep 17 00:00:00 2001
From: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
Date: Mon, 12 Apr 2021 19:08:39 -0700
Subject: [PATCH 1792/1851] soc: xilinx: vcu: update gasket isolation

commit 77883b257c27f47445381c70c9236ade8f87eeaa from
https://github.com/Xilinx/linux-xlnx.git

This patch skips removing vcu gasket isolation in driver if vcu
logiCORE IP removes gasket isolation already.

Signed-off-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/soc/xilinx/xlnx_vcu_core.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/xilinx/xlnx_vcu_core.c b/drivers/soc/xilinx/xlnx_vcu_core.c
index 0415b283c133..9a700e1d59d9 100644
--- a/drivers/soc/xilinx/xlnx_vcu_core.c
+++ b/drivers/soc/xilinx/xlnx_vcu_core.c
@@ -98,7 +98,13 @@ static int xvcu_core_probe(struct platform_device *pdev)
 		dev_warn(&pdev->dev, "No reset gpio info from dts for vcu. This may lead to incorrect functionality if VCU isolation is removed post initialization.\n");
 	}
 
-	iowrite32(VCU_GASKET_VALUE, xvcu->logicore_reg_ba + VCU_GASKET_INIT);
+	/*
+	 * skip sw isolation step if vcu ip has removed gasket isolation
+	 * already
+	 */
+	if (!of_property_read_bool(pdev->dev.of_node, "xlnx,skip-isolation"))
+		iowrite32(VCU_GASKET_VALUE,
+			  xvcu->logicore_reg_ba + VCU_GASKET_INIT);
 
 	ret = mfd_add_devices(&pdev->dev, PLATFORM_DEVID_NONE, xvcu_devs,
 			      ARRAY_SIZE(xvcu_devs), NULL, 0, NULL);
-- 
2.31.1

